com.ibm.streamsx.topology/opt/python/packages/streamsx/topology/graph.py

```diff
@@ -188,3 +191,44 @@ def getSPLOutputPort(self):
         _oport["name"] = self.name
         _oport["connections"] = [port.name for port in self.inputPorts]
         return _oport
+
+class SPLMarker(object):
+
+    def __init__(self, index, kind, name, params, graph):
+        self.index = index
+        self.kind = kind
+        self.name = name
+        self.params = {}
+        self.setParameters(params)
+        self._addOperatorFunction(self.function)
+        self.graph = graph
+
+        self.inputPorts = []
+        self.outputPorts = []
+
+    def generateSPLOperator(self):
+        _op = {}
+        _op["name"] = self.name
+
+        _op["kind"] = self.kind
+        _op["partitioned"] = False
+
+        _op["marker"] = True
+        _op["model"] = "spl"
+        _op["language"] = "java"
```

Comment:
It's a marker "operator" that doesn't really have a language, the idea is that an operator in the graph has:

* model - the programming model of the operator
* kind - the kind of the operator within the programming model
* language - Implementation language

E.g. at some point a filter in Java would be:
 {topology, filter, java}
or Python
{topology, filter, python}

and later once it was translated into SPL the first might be:
{spl, JavaFunctionalFilter, spl}   // Can't rememeber the actual name of the operator

That's the goal, I think the graph currently dives to quickly into SPL, rather than being a later transformation step.